Mathlib/MeasureTheory/Measure/Portmanteau.lean

```diff
@@ -456,9 +452,60 @@ theorem exists_null_frontiers_thickening (μ : Measure Ω) [SigmaFinite μ] (s :
+then for all closed sets `F` we have the limsup condition `limsup (μs i F) ≤ μ F`. -/
+lemma limsup_measure_closed_le_of_forall_tendsto_measure
+    {Ω ι : Type _} {L : Filter ι} [NeBot L]
+    [MeasurableSpace Ω] [PseudoEMetricSpace Ω] [OpensMeasurableSpace Ω]
+    {μ : Measure Ω} [IsFiniteMeasure μ] {μs : ι → Measure Ω}
+    (h : ∀ {E : Set Ω}, MeasurableSet E → μ (frontier E) = 0 →
+            Tendsto (fun i ↦ μs i E) L (𝓝 (μ E)))
+    (F : Set Ω) (F_closed : IsClosed F) :
+    L.limsup (fun i ↦ μs i F) ≤ μ F := by
+  have ex := exists_null_frontiers_thickening μ F
+  let rs := Classical.choose ex
+  have rs_lim : Tendsto rs atTop (𝓝 0) := (Classical.choose_spec ex).1
+  have rs_pos : ∀ n, 0 < rs n := fun n ↦ ((Classical.choose_spec ex).2 n).1
+  have rs_null : ∀ n, (μ : Measure Ω) (frontier (Metric.thickening (rs n) F)) = 0 :=
+    fun n ↦ ((Classical.choose_spec ex).2 n).2
+  have Fthicks_open : ∀ n, IsOpen (Metric.thickening (rs n) F) :=
+    fun n ↦ Metric.isOpen_thickening
+  have key := fun (n : ℕ) ↦ h (Fthicks_open n).measurableSet (rs_null n)
+  apply ENNReal.le_of_forall_pos_le_add
+  intros ε ε_pos μF_finite
+  have keyB := tendsto_measure_cthickening_of_isClosed (μ := μ) (s := F)
+                ⟨1, ⟨by simp only [gt_iff_lt, zero_lt_one], measure_ne_top _ _⟩⟩ F_closed
+  have nhd : Iio ((μ : Measure Ω) F + ε) ∈ 𝓝 ((μ : Measure Ω) F) := by
+    apply Iio_mem_nhds
+    simpa only [add_zero] using ENNReal.add_lt_add_left μF_finite.ne (ENNReal.coe_pos.mpr ε_pos)
+  specialize rs_lim (keyB nhd)
+  simp only [mem_map, mem_atTop_sets, ge_iff_le, mem_preimage, mem_Iio] at rs_lim
+  obtain ⟨m, hm⟩ := rs_lim
+  have aux' := fun i ↦ measure_mono (μ := μs i) (Metric.self_subset_thickening (rs_pos m) F)
+  have aux : (fun i ↦ ((μs i : Measure Ω) F))
+              ≤ᶠ[L] (fun i ↦ (μs i : Measure Ω) (Metric.thickening (rs m) F)) := by
+    exact eventually_of_forall aux'
+  refine (limsup_le_limsup aux).trans ?_
+  rw [Tendsto.limsup_eq (key m)]
+  apply (measure_mono (Metric.thickening_subset_cthickening (rs m) F)).trans (hm m rfl.le).le
 
---namespace
+/-- One implication of the portmanteau theorem:
+Assuming that for all Borel sets `E` whose boundary `∂E` carries no probability mass under a
+candidate limit probability measure `μ` we have convergence of the measures `μs i E` to `μ E`,
+then for all open sets `G` we have the limsup condition `μ G ≤ liminf (μs i G)`. -/
+lemma le_liminf_measure_open_of_forall_tendsto_measure
+    {Ω ι : Type _} {L : Filter ι} [NeBot L]
+    [MeasurableSpace Ω] [PseudoEMetricSpace Ω] [OpensMeasurableSpace Ω]
+    {μ : Measure Ω} [IsProbabilityMeasure μ] {μs : ι → Measure Ω} [∀ i, IsProbabilityMeasure (μs i)]
+    (h : ∀ {E : Set Ω}, MeasurableSet E → (μ : Measure Ω) (frontier E) = 0 →
+            Tendsto (fun i ↦ (μs i : Measure Ω) E) L (𝓝 ((μ : Measure Ω) E)))
+    (G : Set Ω) (G_open : IsOpen G) :
+    (μ : Measure Ω) G ≤ L.liminf (fun i ↦ (μs i : Measure Ω) G) := by
```

Comment:
You can simply write `μs i E` instead of `(μs i : Measure Ω) E`.
```suggestion
lemma le_liminf_measure_open_of_forall_tendsto_measure
    {Ω ι : Type _} {L : Filter ι} [NeBot L]
    [MeasurableSpace Ω] [PseudoEMetricSpace Ω] [OpensMeasurableSpace Ω]
    {μ : Measure Ω} [IsProbabilityMeasure μ] {μs : ι → Measure Ω} [∀ i, IsProbabilityMeasure (μs i)]
    (h : ∀ {E}, MeasurableSet E → μ (frontier E) = 0 → Tendsto (fun i ↦ μs i E) L (𝓝 (μ E)))
    (G : Set Ω) (G_open : IsOpen G) :
    μ G ≤ L.liminf (fun i ↦ μs i G) := by
```